medusa/providers/torrent/json/t411.py

```diff
@@ -119,13 +119,15 @@ def parse(self, data, mode):
         """
         items = []
 
-        unsorted_torrent_rows = data.get('torrents') if mode != 'RSS' else data
-
-        if not unsorted_torrent_rows or not isinstance(unsorted_torrent_rows, dict):
+        # Data is `dict` when episode search and `list` when RSS search
+        if not data or not ((isinstance(data, dict) and data.get('torrents')) or
+                            isinstance(data, list) and len(data) > 0):
```

Comment:
See my previous comments as to the correct fix.  That aside, here you are checking `len(data) > 0` which first of all could be simplified to `len(data)` since that evaluates to `False` if `data == 0`.  Secondly, it would never be false at this point anyway since you already checked `if not data`.